tests/cpp/multidevice.cpp

```diff
@@ -144,13 +145,7 @@ at::Tensor MultiDeviceTest::shardTensor(
   auto stride = extent / nslices;
   // TODO: returning slice 0 temporarily when device is not in the mesh.
   i = (i < 0) ? 0 : i;
-  auto slice = tensor.slice(axis, i * stride, (i + 1) * stride).contiguous();
-  // Temporary until https://github.com/NVIDIA/Fuser/issues/2563. Adds DIDx
-  // axis in front representing the sharded extent of the tensor.
```

Comment:
Because loop-split cases don't need the unsqueeze, I moved the unsqueeze to the users, all of which are in distributed transformer tests. I don't feel the extra typing too bothersome, but I could make `shardTensor(...).unsqueeze(0)` a separate helper with a different name.

Reply:
Sounds good! This will also help flag tests that will eventually be cleaned up.